build-tools/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -77,28 +83,30 @@ public void apply(Project project) {
 
     private void setupDistributionContainer(Project project, Property<Boolean> dockerAvailable) {
         distributionsContainer = project.container(ElasticsearchDistribution.class, name -> {
-            Configuration fileConfiguration = project.getConfigurations().create("es_distro_file_" + name);
+            Configuration fileConfiguration = project.getConfigurations().create(DISTRO_CONFIG_PREFIX + name);
             Configuration extractedConfiguration = project.getConfigurations().create(DISTRO_EXTRACTED_CONFIG_PREFIX + name);
             extractedConfiguration.getAttributes()
                 .attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, ArtifactTypeDefinition.DIRECTORY_TYPE);
+            DependencyHandler dependencies = project.getDependencies();
+
+            ConfigurableFileCollection from = project.getObjects().fileCollection().from(fileConfiguration);
+            ConfigurableFileCollection from2 = project.getObjects().fileCollection().from(extractedConfiguration);
+
             return new ElasticsearchDistribution(
                 name,
                 project.getObjects(),
                 dockerAvailability,
-                fileConfiguration,
-                extractedConfiguration,
-                (dist) -> finalizeDistributionDependencies(project, dist)
```

Comment:
referencing project in a lambda lets the configuration cache choke when trying to serialise this lambda